src/scenes/Authentication/Register/RegisterForm.tsx

```diff
@@ -105,17 +119,32 @@ export const RegisterForm = ({ className, ...props }: RegisterFormProps) => {
   );
 };
 
-const passwordMatching = ({
+const validateFields = ({
   password,
   confirmPassword,
+  realFirstName,
+  realLastName,
+  displayFirstName,
+  displayLastName,
   ..._otherVals
 }: RegisterInput) => {
-  return password && confirmPassword && password !== confirmPassword
-    ? {
-        password: 'Passwords must match',
-        confirmPassword: 'Passwords must match',
-      }
-    : undefined;
+  const errors: Partial<RegisterInput> = {};
+
+  if (!password && confirmPassword && password === confirmPassword) {
+    errors.password = 'Passwords must match';
+    errors.confirmPassword = 'Passwords must match';
+  }
+  if (!realFirstName || realFirstName.length < 2)
+    errors.realFirstName = 'First Name must be longer than 2 characters';
+  if (!realLastName || realLastName.length < 2)
+    errors.realLastName = 'Last Name must be longer than 2 characters';
+  if (!displayFirstName || displayFirstName.length < 2)
+    errors.displayFirstName =
+      'Display First Name must be longer than 2 characters';
+  if (!displayLastName || displayLastName.length < 2)
+    errors.displayLastName =
+      'Display Last Name must be longer than 2 characters';
```

Comment:
These should be done with field validators. See functions in this file and their usages https://github.com/SeedCompany/cord-field/blob/636c73f880d1dd864319aae3c6640db4ae3eb10e/src/components/form/validators.ts

Reply:
Done